IdentityCore/src/webview/embeddedWebview/challangeHandlers/MSIDPKeyAuthHandler.m

```diff
@@ -68,11 +69,16 @@ + (BOOL)handleChallenge:(NSString *)challengeUrl
     
     // Attach client version to response url
     NSURLComponents *responseUrlComp = [[NSURLComponents alloc] initWithURL:[NSURL URLWithString:submitUrl] resolvingAgainstBaseURL:NO];
-    NSMutableArray *queryItems = responseUrlComp.queryItems ? [responseUrlComp.queryItems mutableCopy] : [NSMutableArray new];
-    [queryItems addObject:[[NSURLQueryItem alloc] initWithName:MSID_VERSION_KEY value:MSIDDeviceId.deviceId[MSID_VERSION_KEY]]];
-    responseUrlComp.queryItems = queryItems;
+    NSMutableDictionary *queryDict = [NSMutableDictionary new];
     
-    NSMutableURLRequest *responseReq = [[NSMutableURLRequest alloc]initWithURL:responseUrlComp.URL];
+    for (NSURLQueryItem *item in responseUrlComp.queryItems)
+    {
+        [queryDict setValue:item.value forKey:item.name];
+    }
+    [queryDict setValue:MSIDDeviceId.deviceId[MSID_VERSION_KEY] forKey:MSID_VERSION_KEY];
+    responseUrlComp.percentEncodedQuery = [queryDict msidURLFormEncode];
```

Comment:
Just for my own knowledge:
1. What's the difference between setting percentEncodedQuery and doing percent encoding myself vs setting query without percent encoding.
2. Please note that if we make any mistake in percent encoding (or percent encoding rules are different for Apple and us), this will throw an exception:

`// Getting these properties retains any percent encoding these components may have. Setting these properties assumes the component string is already correctly percent encoded. Attempting to set an incorrectly percent encoded string will cause an exception. Although ';' is a legal path character, it is recommended that it be percent-encoded for best compatibility with NSURL (-stringByAddingPercentEncodingWithAllowedCharacters: will percent-encode any ';' characters if you pass the URLPathAllowedCharacterSet).`

Do we need to worry about that?

Reply:
1. the submitUrl we need is already encoded, and we are using that to begin with. 
If all done right, it'd be the same but we are already working from encoded, thus .percentEncodedQuery is the right call here.

2. This really shouldn't happen, unless our encoding is wrong.

Reply:
👍 thanks for answering :)